src/main/scala/com/touk/hiring_task/multiplex/routes/Routes.scala

```diff
@@ -0,0 +1,89 @@
+package com.touk.hiring_task.multiplex.routes
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+import akka.http.scaladsl.server.directives.RouteDirectives._
+import akka.http.scaladsl.server._
+import com.touk.hiring_task.multiplex.commons._
+import com.touk.hiring_task.multiplex.routes.ErrorCode.ErrorCode
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import io.circe.generic.auto.exportEncoder
+import io.circe.{Decoder, Encoder}
+
+import scala.concurrent.Future
+
+trait Routes extends FailFastCirceSupport {
+
+  protected def completeEitherOrError[A](fe: Future[Either[MultiplexException, A]], returnBody: Boolean = true)(implicit marshaller: ToEntityMarshaller[A]): Route =
+    Directives.onSuccess(fe) {
+      case Right(a) => if (returnBody) complete(StatusCodes.OK, a) else complete(StatusCodes.OK)
+      case Left(t) => completeHttpError(t)
+    }
+
+  protected def completeError(httpError: HttpErrorDto) =
+    complete(httpError.status, httpError.error)
+
+  protected def completeHttpError(ex: MultiplexException): StandardRoute =
+    ex match {
```

Comment:
do you think that changing domain Error to API error in one place is a good idea? why?